pkg/apis/lighthouse.submariner.io/v2alpha1/types.go

```diff
@@ -69,3 +69,108 @@ type ServiceExportList struct {
 	metav1.ListMeta `json:"metadata"`
 	Items           []ServiceExport `json:"items"`
 }
+
+//+genclient
+//+k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +genclient:noStatus
+
+// ServiceImport describes a service imported from clusters in a supercluster.
+type ServiceImport struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// +optional
+	Spec ServiceImportSpec `json:"spec,omitempty"`
+	// +optional
+	Status ServiceImportStatus `json:"status,omitempty"`
+}
+
+// ServiceImportType designates the type of a ServiceImport
+type ServiceImportType string
+
+const (
+	// SuperclusterIP are only accessible via the Supercluster IP.
+	SuperclusterIP ServiceImportType = "SuperclusterIP"
+	// Headless services allow backend pods to be addressed directly.
+	Headless ServiceImportType = "Headless"
+)
+
+// ServiceImportSpec describes an imported service and the information necessary to consume it.
+type ServiceImportSpec struct {
+	// +listType=atomic
+	Ports []ServicePort `json:"ports"`
+	// +optional
+	IP string `json:"ip,omitempty"`
+	// +optional
+	Type ServiceImportType `json:"type"`
+	// +optional
+	SessionAffinity corev1.ServiceAffinity `json:"sessionAffinity"`
+	// +optional
+	SessionAffinityConfig *corev1.SessionAffinityConfig `json:"sessionAffinityConfig"`
+	// +optional
+	Items []ClusterServiceInfo `json:"clusterServiceInfo"`
+}
+
+// ServicePort represents the port on which the service is exposed
+type ServicePort struct {
+	// The name of this port within the service. This must be a DNS_LABEL.
+	// All ports within a ServiceSpec must have unique names. When considering
+	// the endpoints for a Service, this must match the 'name' field in the
+	// EndpointPort.
+	// Optional if only one ServicePort is defined on this service.
+	// +optional
+	Name string `json:"name,omitempty"`
+
+	// The IP protocol for this port. Supports "TCP", "UDP", and "SCTP".
+	// Default is TCP.
+	// +optional
+	Protocol corev1.Protocol `json:"protocol,omitempty"`
+
+	// The application protocol for this port.
+	// This field follows standard Kubernetes label syntax.
+	// Un-prefixed names are reserved for IANA standard service names (as per
+	// RFC-6335 and http://www.iana.org/assignments/service-names).
+	// Non-standard protocols should use prefixed names such as
+	// mycompany.com/my-custom-protocol.
+	// Field can be enabled with ServiceAppProtocol feature gate.
+	// +optional
+	AppProtocol *string `json:"appProtocol,omitempty"`
+
+	// The port that will be exposed by this service.
+	Port int32 `json:"port"`
+}
+
+//ClusterServiceInfo provides information about service in each cluster.
+type ClusterServiceInfo struct {
+	//The unique identity of the cluster where the service is available. This field is
+	//not optional.
```

Comment:
```suggestion
	// The unique identity of the cluster where the service is available. This field is
	// not optional.
```

For readability consistency - also in other places.